src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -292,6 +292,10 @@ public final void removeInterestConnect(SelectionKey key) {
      * @param key the key.
      */
     public final void addInterestRead(SelectionKey key) {
+        if (LOGGER.isLoggable(Level.FINEST)) {
```

Comment:
@stephenc so the performance without inlining didn;t seem to bad - and if we leave it as is then you can use the standard Jenkins UI (or JMX for agents) to change the logging level